Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -85,27 +87,30 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
                 sendApplicationStartAction()
             }
             needsViewUpdate = true
+        case let command as RUMStartViewCommand where command.identity !== identity:
+            isActiveView = false
         case let command as RUMStopViewCommand where command.identity === identity:
+            isActiveView = false
```

Comment:
let's say the app pushes `VC1` and presents `VC2` and then dismisses `VC2`:
1. `startView(VC1)` (active view is VC1)
2. `startView(VC2)` (active view is VC2)
3. `stopView(VC2)` (shouldn't active view be VC1?)

if i understand well, currently in this case we are left with no active view, correct?

Reply:
This would mean the incorrect instrumentation. When a new View is started, the old one should be stopped. Transferring it to the `UIKit` language: when presenting `VC2`, the `VC1` should be stopped, and when dismissing the `VC2`, `VC1` should be started again.

We know this is not the most convenient way for `UIKit`, but what we do here is complying to the RUM product requirements, not `UIKit` specifics. The RUM requirements are loosely defined, so they leave the door open for any integration specifics (`UIKit`, `SwiftUI`, Android activities & fragments, web frameworks, ...). This sort of problems will be solved later with specific `UIKit` helpers and in the auto instrumentation components. It should not impact the general rules of the RUM.